arroyo-sql/src/expressions.rs

```diff
@@ -2593,3 +2604,104 @@ impl CaseExpression {
+}
+
+fn convert_expression_to<T>(expr: Expression, expr_name: &str) -> Result<T, anyhow::Error>
+where
+    T: for<'a> TryFrom<&'a str, Error = String>,
+{
+    let Expression::Literal(LiteralExpression{literal: ScalarValue::Utf8(Some(literal_string))}) = expr else {
+        bail!(format!("Can only convert a literal into {}",expr_name))
+    };
+    literal_string
+        .as_str()
+        .try_into()
+        .map_err(anyhow::Error::msg)
+}
+
+impl TryFrom<(BuiltinScalarFunction, Vec<Expression>)> for DateTimeFunction {
+    type Error = anyhow::Error;
+
+    fn try_from(
+        value: (BuiltinScalarFunction, Vec<Expression>),
+    ) -> std::result::Result<Self, Self::Error> {
+        let func = value.0;
+        let mut args = value.1;
+        match (args.len(), func) {
+            (2, BuiltinScalarFunction::DatePart) => {
+                let arg1 = args.remove(0);
+                let arg2 = Box::new(args.remove(0));
+                let date_part = convert_expression_to::<DatePart>(arg1, "date_part")?;
+                Ok(DateTimeFunction::DatePart(arg2, Box::new(date_part)))
+            }
+            (2, BuiltinScalarFunction::DateTrunc) => {
+                let arg1 = args.remove(0);
+                let arg2 = Box::new(args.remove(0));
+                let date_trunc_precision =
+                    convert_expression_to::<DateTruncPrecision>(arg1, "date_trunc")?;
+                Ok(DateTimeFunction::DateTrunc(
+                    arg2,
+                    Box::new(date_trunc_precision),
+                ))
+            }
+            (_, func) => bail!("function {} with args {:?} not supported", func, args),
+        }
+    }
+}
+
+fn enum_to_syn<T: serde::Serialize + std::fmt::Debug>(t: &T, prefix: &str) -> syn::Expr {
+    let s = serde_json::to_string(t).unwrap();
+    let s = format!("{}::{}", prefix, &s[1..s.len() - 1]); // Remove the quotation marks
+    syn::parse_str(&s).unwrap()
+}
+
+impl DateTimeFunction {
+    fn non_null_function_invocation(&self) -> syn::Expr {
+        match self {
+            DateTimeFunction::DateTrunc(_, _) => {
+                parse_quote!(arroyo_worker::operators::functions::datetime::date_trunc(
+                    arg1, arg2
+                ))
+            }
+            DateTimeFunction::DatePart(_, _) => {
+                parse_quote!(arroyo_worker::operators::functions::datetime::date_part(
+                    arg1, arg2
+                ))
+            }
+        }
+    }
+
+    fn to_syn_expression(&self) -> syn::Expr {
+        let function = self.non_null_function_invocation();
+        let (expr1, expr2) = match self {
+            DateTimeFunction::DatePart(arg1, arg2) => {
+                (arg1.to_syn_expression(), enum_to_syn(arg2, "DatePart"))
+            }
+            DateTimeFunction::DateTrunc(arg1, arg2) => (
+                arg1.to_syn_expression(),
+                enum_to_syn(arg2, "DateTruncPrecision"),
+            ),
+        };
```

Comment:
I'd recommend rather than the intricacies of `enum_to_syn()`, just using the debug formatting. Something like
```suggestion
                let (expr1, expr2) : (syn::Expr, syn::Expr) = match self {
            DateTimeFunction::DatePart(arg1, arg2) => {
                let arg2 = format!("arroyo_types::DatePart::{:?}", arg2);
                (arg1.to_syn_expression(), parse_str(&arg2).unwrap())
            }
            DateTimeFunction::DateTrunc(arg1, arg2) => {
                let arg2 = format!("arroyo_types::DateTruncPrecision::{:?}", arg2);
                (arg1.to_syn_expression(), parse_str(&arg2).unwrap())
            },
        };    
```

Reply:
Addressed with https://github.com/ArroyoSystems/arroyo/pull/181/commits/764bd3d36a3117a579589c4a7d73cae634f3db81